### PR TITLE
Octavia l7 rule support - part 5(final): update

### PR DESCRIPTION
**Pull request by lingxiankong**
For #832

Octavia l7 rule PUT API implementation:

- https://github.com/openstack/octavia/blob/69a45c254be854dbdbff946dbe2564711cdecec3/octavia/api/v2/controllers/l7rule.py#L189

### CODE REVIEW

**Comment by lingxiankong** [comment]
@jtopjian @jrperritt the last one patch for review

**Comment by jtopjian** [comment]
@lingxiankong It looks like `AdminStateUp` is a valid field:

https://github.com/openstack/octavia/blob/69a45c254be854dbdbff946dbe2564711cdecec3/octavia/api/v2/types/l7rule.py#L83

I reviewed https://github.com/gophercloud/gophercloud/pull/924 wondering how we missed that one earlier and saw that you linked to the `v1/types/l7rule.py` file. :)

https://github.com/openstack/octavia/blob/86812e1309feb1bbfbc5448c12b80602d2d0a141/octavia/api/v1/types/l7rule.py#L31

I think the addition of this field in both `CreateOpts` and `UpdateOpts` is small enough to add to this PR without causing too much complication. Or you can create a new PR for it -- your choice.

The field definition would be the same for both `CreateOpts` and `UpdateOpts`:

```go
AdminStateUp *bool `json:"admin_state_up,omitempty"`
```

Other than that, this looks good to me :)

**Comment by jtopjian** [comment]
@lingxiankong Again, commented too early - sorry. It looks like `AdminStateUp` is missing for both Policy and Rule.

It's best to make a separate PR at this point. You can use #832 as the referenced issue -- no need to make a new issue.

The PR will simply include the missing `AdminStateUp` struct for the 4 structs.

File: openstack/loadbalancer/v2/l7policies/requests.go [hunk 1]
```diff
@@ -279,3 +279,42 @@ func DeleteRule(c *gophercloud.ServiceClient, policyID string, ruleID string) (r
...
+}
+
+// ToRuleUpdateMap builds a request body from UpdateRuleOpts.
+func (opts UpdateRuleOpts) ToRuleUpdateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "rule")
+}
+
+// UpdateRule allows Rule to be updated.
+func UpdateRule(c *gophercloud.ServiceClient, policyID string, ruleID string, opts UpdateRuleOptsBuilder) (r UpdateRuleResult) {
+	b, _ := opts.ToRuleUpdateMap()
```
Comment (jtopjian) [inline openstack/loadbalancer/v2/l7policies/requests.go]: I was about to merge this and caught this. I see it's also done with the `Update` function which I didn't catch earlier either.

Granted it's really hard to trigger an error with this, we should still check for an error regardless.

```go
b, err := opts.ToRuleUpdateMap()
if err != nil {
  r.Err = err
  return
}
```

**Comment by jtopjian** [comment]
LGTM - thank you for the update :)

**Pull request merged**